core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
         echo "Nothing to delete."
     fi
-    kubectl create -f - <<EOF
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+}
  
-        echo "Delete the kubeflex provider object if it already exists, and then create it."
-        if ! kubectl delete spaceproviderdesc pkflex ; then
-            echo "Nothing to delete."
-        fi
-        kubectl create -f - <<EOF
+function create_kubeflex_provider() {
+    echo "Waiting for the kubeflex provider to be ready... this may take a while."
+    (
+        until [ "$(kubectl --kubeconfig $host_kubeconfig get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
+            sleep 10
+        done
+    )
+
+    echo "Delete the kubeflex provider object if it already exists, and then create it."
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete spaceproviderdesc $PROVIDER_NAME ; then
+        echo "Nothing to delete."
+    fi
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkflex
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kubeflex"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: corecluster
 EOF
-    fi
-
-    KUBECONFIG=$oldKubeconfig
+    echo "Waiting for default spaceprovider to reach the Ready phase."
+    until [ "$(kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get spaceproviderdesc $PROVIDER_NAME -o yaml | grep Ready )" != "" ]; do
+        sleep 1
+    done 
 }
 
 
-function wait_kcp_ready() {
-    echo "Waiting for kcp to be ready... this may take a while."
+function set_provider_adapters() {
+    echo "Waiting for space manager to be ready... this may take a while."
     (
-        KUBECONFIG=
-        # while ! kubectl exec $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp -- ls /home/kubestellar/ready &> /dev/null; do
-        #    sleep 10
-        # done
-        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
-           sleep 10
+        until [ "$(kubectl --kubeconfig $host_kubeconfig logs $(kubectl --kubeconfig $host_kubeconfig get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c space-manager | grep '***READY***')" != "" ]; do
```

Comment:
Looking for log contents is fragile, it stops working after the log rolls. Could this code instead look for some persistent state that the Space Manager establishes?

Reply:
Same as the comment above.  This is existing code that is already running in production.  But your change proposal totally makes sense and is the right thing to do.  I'll create an issue for it and follow up.

Reply:
https://github.com/kubestellar/kubestellar/issues/1332